frontend/src/lib/k8s/cluster.ts

```diff
@@ -679,37 +707,49 @@ export function makeKubeObject<T extends KubeObjectInterface | KubeEvent>(
         resourceAttrs['resource'] = this.pluralName;
       }
 
-      const spec = {
-        resourceAttributes: resourceAttrs,
-      };
+      // @todo: We should get the API info from the API endpoint.
+
+      // If we already have the group, version, and resource, then we can make the request
+      // without trying the API info, which may have several versions and thus be less optimal.
+      if (!!resourceAttrs.group && !!resourceAttrs.version && !!resourceAttrs.resource) {
+        return this.fetchAuthorization(resourceAttrs);
+      }
+
+      // If we don't have the group, version, and resource, then we have to try all of the
+      // API info versions until we find one that works.
+      const apiInfo = this.apiEndpoint.apiInfo;
+      for (let i = 0; i < apiInfo.length; i++) {
+        const { group, version, resource } = apiInfo[i];
+
+        if (!!resourceAttrs.resource) {
```

Comment:
Could this use resourceAttrs from the previous iteration of the loop? If group is set in i=0, and then not in i=1... it will use the group from i=0;. Is this intentional?

Reply:
Hmm, that would be a possible side effect if the apiInfo doesn't have all the values in all iterations. But I think it always does. I will try to refine the logic nonetheless.